front/temporal/hard_delete/activities.ts

```diff
@@ -0,0 +1,118 @@
+import { Context } from "@temporalio/activity";
+import { QueryTypes, Sequelize } from "sequelize";
+
+import config from "@app/lib/production_checks/config";
+import logger from "@app/logger/logger";
+import type {
+  RunExecutionRow,
+  RunsJoinsRow,
+} from "@app/temporal/hard_delete/types";
+import {
+  getRunExecutionsDeletionCutoffDate,
+  isSequelizeForeignKeyConstraintError,
+} from "@app/temporal/hard_delete/utils";
+
+const BATCH_SIZE = 100;
+
+export async function purgeExpiredRunExecutions() {
+  const primaryCoreDatabaseUri = config.getPrimaryCoreDatabaseUri();
+  const coreSequelize = new Sequelize(primaryCoreDatabaseUri, {
+    logging: false,
+  });
+
+  const cutoffDate = getRunExecutionsDeletionCutoffDate();
+
+  logger.info(
+    {},
+    `About to purge block and run executions anterior to ${new Date(
+      cutoffDate
+    ).toISOString()}.`
+  );
+
+  let hasMoreRunsToPurge = true;
+  let runsPurgedCount = 0;
+  do {
```

Comment:
It was not possible to have one activity to fetch the next ids to delete an another one to delete them? (As opposed to having one big activity with a 60min timeout?)

Reply:
We could, but this should be pretty efficient, the 60 minutes will be downgrade to 5 minutes once the first batch is done. Let me add a TODO.

And we don't even care about next id, given that we delete rows we read, we can always use the same SQL query 😉 .